\begin{center}
Gabe Morris
\end{center}

In [1]:
# Notebook Preamble
import matplotlib.pyplot as plt
import numpy as np

plt.style.use('maroon_ipynb.mplstyle')

\pagebreak
\tableofcontents
\pagebreak

\begin{center}
\begin{tabular}{c c c}
ME 8253 & Homework 6 & Gabe Morris \\
& & gnm54
\end{tabular}
\end{center}

#
## Given
A stepped circular rod of 4340 steel (with $S_u$ = 1468 MPa) with diameters of 60 and 45 mm has a root radius of 3 mm at the stepped section. The rod is to be subjected to axial cyclic loading.

![image not found](images/f1.PNG)

The cyclic yield strength ($S^\prime_y$) is estimated from:

$$S^\prime_y=K^\prime(0.002)^{n^\prime}$$

where $K^\prime$ and $n^\prime$ are given in Table A.2

For the purpose of constructing Haigh diagram, exact value of $\sigma_f$ is not needed, as the diagram is not very sensitive to its value. Since $\sigma_f$ is not listed in Table A.2 for this material, we use Eq. 5.20 in the textbook to approximate it as

$$\sigma_f\approx S_u + 345\ (MPa)$$

## Find
Using a Haigh diagram, determine the following for an approximate median fatigue life of $10^6$ cycles:

a. What fully reversed alternating force, $P_a$, can be applied?
b. What is the maximum value of $P_a$, if proper compressive residual stresses are present at the notch root? What is the magnitude of the compressive residual stress needed to obtain this maximum alternating stress?
c. What value of $P_a$ can be applied if the residual stress calculated in (b) is tensile? What fully reversed alternating force, $P_a$, can be applied?

## Solution
According to equation 4.3b, the endurance limit is $700\ MPa$ for materials with an ultimate strength greater than $1400\ MPa$. With the size effect, the endurance limit becomes

$$S_f=0.85(700)=595\ MPa$$

### Part A

![image not found](images/f2.png)

In [2]:
D, d, r = 60, 45, 3  # mm
Su = 1468  # MPa
size_effect = 0.85

if Su <= 1400:
    Sf = 0.5*Su*size_effect
else:
    Sf = 700*size_effect

D/d

1.3333333333333333

In [3]:
r/d

0.06666666666666667

From above the stress concentration factor is $K_t=2$. The fatigue notch factor for the fully reversed condition is

$$K_f=1+\frac{K_t-1}{1+a/r}$$

where $a=0.0254\left(\frac{2070}{S_u}\right)^{1.8}$ with $a$ in mm and $S_u$ in MPa.

In [4]:
Kt = 2
a = 0.0254*(2070/Su)**1.8
a  # mm

0.047149103389883054

In [5]:
Kf = 1 + (Kt - 1)/(1 + a/r)
Kf

1.9845268144780213

Since the stress is maximized at the smaller diameter, 

$$P_a=\frac{S_f}{K_f}\left(\frac{\pi}{4}\right)(d)^2$$

In [6]:
Pa = Sf/Kf*np.pi/4*d**2
Pa  # N

476842.4418454644

### Part B
From Table A.2,

| Property     | Value     |
|--------------|-----------|
| $S_y$        | 1371  MPa |
| $S^\prime_y$ | 863  MPa  |
| $S_f$        | 595   MPa |
| $\sigma_f$   | 1813  MPa |
| $K_f$        | 1.98      |
| $S_{cat}$    | 70    MPa |

$S_{cat}$ comes from the fact that this is a hard steel. The other calculations are shown below.

In [7]:
sig_f = Su + 345
sig_f  # MPa

1813

In [8]:
K_prime, n_prime = 1996, 0.135
Sy_prime = K_prime*0.002**n_prime
Sy_prime  # MPa

862.5804014077875

![image not found](images/f3.png)

In [9]:
Sa = 460  # MPa
Pa = Sa*np.pi/4*d**2
Pa  # N

731598.3892047232

### Part C
From above, point B is when the mean stress is tensile.

In [10]:
Sa = 90
Pa = Sa*np.pi/4*d**2
Pa  # N

143138.81527918496